File: app/src/main/java/com/respondingio/battlegroundsbuddy/weapons/HomeWeaponsFragment.java
package com.respondingio.battlegroundsbuddy.weapons;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.respondingio.battlegroundsbuddy.R;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.Set;

import butterknife.ButterKnife;

import static android.content.Context.MODE_PRIVATE;

public class HomeWeaponsFragment extends Fragment {

    class ViewPagerAdapter extends FragmentPagerAdapter {

        //"Teaser \uD83D\uDE08"

        ViewPagerAdapter(FragmentManager manager) {
            super(manager);
        }

        @Override
        public int getCount() {
            return mStringArray.size();
        }

        @Override
        public Fragment getItem(int position) {
            Fragment currentFragment;
            Bundle bundle;
            currentFragment = new MainWeaponsList();
            bundle = new Bundle();

            if (tabLayout.getTabCount() == 10 && position == 0) {
                bundle.putBoolean("isFavoriteTab", true);
            } else if (tabLayout.getTabCount() == 10) {
                bundle.putInt("pos", position - 1);
            } else {
                bundle.putInt("pos", position);
            }

            currentFragment.setArguments(bundle);
            return currentFragment;
        }

        @Override
        public CharSequence getPageTitle(int position) {
            return mStringArray.get(position);
        }
    }

    private List<String> mStringArray = new ArrayList<>();

    private TabLayout tabLayout;

    private String title[] = {"Assault Rifles", "Sniper Rifles", "SMGs", "Shotguns", "Pistols", "LMGs",
            "Throwables", "Melee", "Misc"};

    @Nullable
    @Override
    public View onCreateView(@NonNull final LayoutInflater inflater, @Nullable final ViewGroup container,
            @Nullable final Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(R.layout.fragment_home_weapons, container, false);
        ButterKnife.bind(this, view);

        mStringArray.addAll(Arrays.asList(title));

        final ViewPager viewPager = view.findViewById(R.id.viewpager);
        ViewPagerAdapter adapter = new ViewPagerAdapter(getChildFragmentManager());
        viewPager.setAdapter(adapter);

        tabLayout = view.findViewById(R.id.tabs);
        tabLayout.setupWithViewPager(viewPager);

        if (getActivity() != null) {
            final SharedPreferences sharedPreferences = getActivity().getSharedPreferences("com.austinhodak.pubgcenter", MODE_PRIVATE);
            Set<String> favs = sharedPreferences.getStringSet("favoriteWeapons", null);
            if (favs != null && !favs.isEmpty()) {
                tabLayout.addTab(tabLayout.newTab().setCustomView(R.layout.tab_fav), 0);

                tabLayout.setScrollPosition(1, 0f, true);
                viewPager.setCurrentItem(1);

                int emoji = 0x2B50;

                mStringArray.add(0, new String(Character.toChars(emoji)));
                adapter.notifyDataSetChanged();
            }
        }

        return view;
    }
}

File: app/src/main/java/com/respondingio/battlegroundsbuddy/weapons/HomeWeaponsFragment.kt
package com.respondingio.battlegroundsbuddy.weapons

import android.content.Context.MODE_PRIVATE
import android.content.SharedPreferences
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.google.android.gms.ads.AdListener
import com.respondingio.battlegroundsbuddy.R
import com.respondingio.battlegroundsbuddy.utils.Premium
import kotlinx.android.synthetic.main.fragment_home_weapons.*
import java.util.*

class HomeWeaponsFragment : Fragment() {

    private val mStringArray = ArrayList<String>()



    internal inner class ViewPagerAdapter

    (manager: FragmentManager) : FragmentPagerAdapter(manager) {

        override fun getCount(): Int {
            return mStringArray.size
        }

        override fun getItem(position: Int): Fragment {
            val currentFragment: Fragment
            val bundle = Bundle()
            currentFragment = MainWeaponsList()

            if (tabs?.tabCount == 10 && position == 0) {
                bundle.putBoolean("isFavoriteTab", true)
            } else if (tabs?.tabCount == 10) {
                bundle.putInt("pos", position - 1)
            } else {
                bundle.putInt("pos", position)
            }

            currentFragment.arguments = bundle
            return currentFragment
        }

        override fun getPageTitle(position: Int): CharSequence? {
            return mStringArray[position]
        }
    }

    private var mSharedPreferences: SharedPreferences? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        mSharedPreferences = requireActivity().getSharedPreferences("com.austinhodak.pubgcenter", MODE_PRIVATE)
        return inflater.inflate(R.layout.fragment_home_weapons, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupTabs()

        if (!Premium.isAdFreeUser()) {
            weaponListAd?.adListener = object: AdListener() {
                override fun onAdFailedToLoad(p0: Int) {
                    super.onAdFailedToLoad(p0)
                    weaponListAd?.visibility = View.GONE
                }

                override fun onAdLoaded() {
                    super.onAdLoaded()
                    weaponListAd?.visibility = View.VISIBLE
                }
            }
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        weaponListAd?.destroy()
    }

    private fun setupTabs() {
        val title = arrayOf(
                getString(R.string.assault_rifles),
                getString(R.string.sniper_rifles),
                "SMGs",
                getString(R.string.shotguns),
                getString(R.string.pistols),
                "LMGs",
                getString(R.string.throwables),
                getString(R.string.melee), "Misc"
        )
        mStringArray.addAll(Arrays.asList(*title))

        val adapter = ViewPagerAdapter(childFragmentManager)
        viewpager?.adapter = adapter

        tabs?.setupWithViewPager(viewpager)

        if (activity != null) {
            val sharedPreferences = requireActivity().getSharedPreferences("com.austinhodak.pubgcenter", MODE_PRIVATE)
            val favs = sharedPreferences.getStringSet("favoriteWeapons", null)
            if (favs != null && !favs.isEmpty()) {
                tabs?.addTab(tabs?.newTab()?.setCustomView(R.layout.tab_fav)!!, 0)

                tabs?.setScrollPosition(1, 0f, true)
                viewpager.currentItem = 1

                val emoji = 0x2B50

                mStringArray.add(0, String(Character.toChars(emoji)))
                adapter.notifyDataSetChanged()
            }
        }
    }
}
